pants

```diff
@@ -65,8 +98,14 @@ export PY="${PY:-python3}"
 # Pants will try to use Python 2 for subprocesses. Without setting minor version constraints, when using Python 3
 # we could end up using a different Python minor version for subprocesses than we do for Pants.
 py_major_minor_patch=$(${PY} -c 'import sys; print(".".join(map(str, sys.version_info[0:3])))')
+
 export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="${PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS:-['CPython==${py_major_minor_patch}']}"
 
+# We can currently assume the CI environment does not switch python versions within a shard.
+if [[ "${ONLY_USING_SINGLE_PYTHON_VERSION:-false}" != 'true' ]]; then
+  clean_if_interpreter_changed "$py_major_minor_patch"
```

Comment:
We only need to clean if the interpreter constraints, not if `PY` changed. In the default case, how we have set up `./pants` and `./pants2`, changing `PY` will by consequence change the interpreter constraints, but what we want to monitor is the constraints rather than `PY`.

```suggestion
  clean_if_interpreter_constraints_changed
```

Reply:
Oops, yes, forgot to remove that! Thanks!